samples/javascript_nodejs/50.teams-messaging-extensions-search/bots/teamsMessagingExtensionsSearchBot.js

```diff
@@ -3,22 +3,70 @@
+var fs = require('fs');
+const AdaptiveCard = require('../Resources/RestaurantCard.json');
+const ConnectorCard = require('../Resources/ConnectorCard.json');
+var configuration = require('dotenv').config();
+var env = configuration.parsed;
+var baseurl = env.BaseUrl;
+
+debugger;
+var publicDir = require('path').join(__dirname,'../public/Images'); 
 
 class TeamsMessagingExtensionsSearchBot extends TeamsActivityHandler {
+    
     async handleTeamsMessagingExtensionQuery(context, query) {
         const searchQuery = query.parameters[0].value;
-        const response = await axios.get(`http://registry.npmjs.com/-/v1/search?${ querystring.stringify({ text: searchQuery, size: 8 }) }`);
+      
+        var attachments = [];
+        switch(searchQuery){
+            case 'adaptive card':           
+               var adadptivecardresponse =   this.GetAdaptiveCard();
+               return adadptivecardresponse;
+               
+            case 'connector card':              
+                var connectorcardresponse = this.GetConnectorCard();
+                return connectorcardresponse;
+
+            case `result grid`:              
+                var resultgridresponse = this.GetResultGrid();
+                return resultgridresponse;
+            default: 
+            const response = await axios.get(`http://registry.npmjs.com/-/v1/search?${ querystring.stringify({ text: searchQuery, size: 8 }) }`);
 
+           
+            response.data.objects.forEach(obj => {
+                const heroCard = CardFactory.heroCard(obj.package.name);
+                const preview = CardFactory.heroCard(obj.package.name);
+                preview.content.tap = { type: 'invoke', value: { description: obj.package.description } };
+                const attachment = { ...heroCard, preview };
+                attachments.push(attachment);
+            });
+    
+            return {
+                composeExtension: {
+                    type: 'result',
+                    attachmentLayout: 'list',
+                    attachments: attachments
+                }
+            };
+        }       
+    }
+
+     GetAdaptiveCard() {
+       
+        var filepath = "./Resources/RestaurantCard.json";
         const attachments = [];
-        response.data.objects.forEach(obj => {
-            const heroCard = CardFactory.heroCard(obj.package.name);
-            const preview = CardFactory.heroCard(obj.package.name);
-            preview.content.tap = { type: 'invoke', value: { description: obj.package.description } };
-            const attachment = { ...heroCard, preview };
-            attachments.push(attachment);
-        });
+        const preview = CardFactory.thumbnailCard(
+            'Adaptive Card',
+            'Please select to get the card'
+        );
 
+        const adaptive = CardFactory.adaptiveCard(AdaptiveCard);
+        
+        const attachment = {...adaptive,preview}
+        attachments.push(attachment);
```

Comment:
It looks like this method and `GetConnectorCard()` both only `push()` a single attachment. Please:

1. Remove `const attachments = [];`
2. Remove `attachments.push(attachment);`
3. Adjust the return to:

```js
return {
    composeExtension: {
        type: 'result',
        attachmentLayout: 'list',
        attachments: [attachment]
    }
};
```